string name
---
bool result
string full_name